src/libstd/sys/unix/thread.rs

```diff
@@ -206,6 +207,29 @@ pub unsafe fn create(stack: uint, p: Thunk) -> rust_thread {
     native
 }
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
+pub unsafe fn set_name(name: &str) {
+    // Using prctl() rather than pthread_setname_np(),
+    // because pthread_setname_np() wasn't added until glibc 2.12
+    // PR_SET_NAME since Linux 2.6.9
+    let cname = CString::from_slice(name.as_bytes());
+    prctl(15i32 /* = PR_SET_NAME */, cname.as_ptr() as u64, 0u64, 0u64, 0u64);
```

Comment:
Can this use the [`linkage = "extern_weak"`](https://github.com/rust-lang/rust/blob/777435990e0e91df6b72ce80c9b6fa485eeb5daa/src/libstd/sys/unix/thread.rs#L229-L240) trick instead?